src/app/tasks/file-reencryption-task.ts

```diff
@@ -118,22 +123,76 @@ export class FileReencryptionTask implements Task {
     this._errors.push(STATUS.CANCELED);
     this._status = STATUS.CANCELED;
     this._taskManagerService.onTaskEvent();
+    this._emitFinish(false);
+    this.destroy();
+  }
+
+  onFinish() {
+    return this._finishSubject.asObservable();
+  }
+
+  destroy() {
+    this._unsubscribeTransactionDialog();
+
+    if (this._subscription) {
+      this._subscription.unsubscribe();
+      this._subscription = null;
+    }
   }
 
   async callUserAction(): Promise<any> {
-    try {
-      this._needsUserAction = false;
-      this._status = STATUS.PENDING_FINALISATION;
-      this._taskManagerService.onTaskEvent();
-      await this._dataProductService.finaliseDataProductPurchase(this._dataProductAddress, this._buyerAddress, this._result);
-      this._status = STATUS.FINISHED;
-      this._finished = true;
-      this._taskManagerService.onTaskEvent();
-    } catch (error) {
-      console.warn(error);
-      this._needsUserAction = true;
-      this._status = STATUS.REJECTED;
-      this._taskManagerService.onTaskEvent();
+    return;
+  }
+
+  private _emitFinish(value): void {
+    this._finishSubject.next(value);
+    this._finishSubject.complete();
+  }
+
+  private async _finalise(): Promise<any> {
+    this._status = STATUS.PENDING_FINALISATION;
+    this._taskManagerService.onTaskEvent();
+
+    const transactionDialogRef = this._dialog.open(TransactionDialogComponent, {
+      disableClose: true
+    });
+
+    this._unsubscribeTransactionDialog();
+    this._transactionDialogSubscription = transactionDialogRef.afterClosed().subscribe(result => {
+      if (result) {
+        this._status = STATUS.FINISHED;
```

Comment:
`this._status`, `this._finished`, `this._taskManagerService.onTaskEvent()` please put before `if` clause (no need to repeat in `else`), `this.destroy()` may be after